### DISCUSSION TITLE
Define the path to the _extensions folder; is this possible?

### DISCUSSION THREAD

**Discussion by cavo789** [quarto-dev/quarto-cli#8098] [Q&A] [answered]
### Description

Hi all,

I'm trying to build my own Docker image for Quarto and it's work until now.

What I'm trying now is to put a `quarto add xxx` command in my Dockerfile. The extension is well downloaded but stored in the `/_extensions` root directory in my container. Ok, fine.

But now, when I try to convert a .qmd file, Quarto is displaying an error telling he didn't retrieve the extension.  Ok, so far, I've understood that `_extensions` should be in my project and there, it's not my ambition.  I wish to be able to install the extension in my Docker container so I can use the image for more than one project.

Is there any way to override the location and make it global?

The code below didn't work...

```yaml
filters:
   - /_extensions/quarto-ext/include-code-files
```

Can we f.i. set a global environment variable like `QUARTO_EXTENSIONS_PATH` or anything else?

Many thanks!!!!

**Comment by mcanouil** [comment, marked as answer]
Extensions are not installed globally. They are on a per project basis.
`quarto add` will install the extension in the current directory.
If you don't want that then, install the extension in the proper directory or use hard link for UNIX-based Docker image.

**Reply by cavo789** [reply]
An answer so fast; thanks a lot!

In my scenario, I've a folder called `/input` in my Docker image and I need to run `docker run xxx -v ${PWD}:/input -w /input xxx quarto render` to run my container; map my current folder as `/input` and do the conversion thanks to quarto.

If I correctly understand your suggestion, I should create a Linux hard link in my `/input` folder to the `/_extensions` folder. But in that case, due to the `-v` flag, I'll get that _extensions folder on my host too and I don't want this.

So far, I don't see how I can install extensions in my Docker image and use them *globally*.

I'm quite new to Quarto (and already love it); is there any reason why an extension can't be global?

**Reply by mcanouil** [reply]
> is there any reason why an extension can't be global?

The reason is that Quarto focuses on reproducibility by leveraging the fact that Quarto projects should be self-contained.
This is also why symbolic link are not recommended (and does not work well anyway at the moment), and why most paths starting with `/` inside a Quarto project means "at the root of the project" and not at the root of the "host".

**Reply by cavo789** [reply]
Ok; thanks.

So far I've follow the idea of symlink. I don't really like it but it works.

The basic idea is to check that I've a `_extensions` folder in the root of my Docker image and not in my `input` directory. In that case, I make the symlink, run quarto then remove the link.

It works but, yes, not really bullet-proof.

```bash
local -i link_extension
link_extension=0

set +e

if [[ -d "/_extensions" ]] && [[ ! -d "/input/_extensions" ]]; then
   link_extension=1
   ln -s "/_extensions" "/input/_extensions"
fi

quarto render xxx

[[ $link_extension -eq 1 ]] && rm -rf "/input/_extensions"

set -e
```

Thanks for the idea.
